new code warnings in dutil.c ncio.c test_pathcvt.c tst_diskless6.c tst_quantize.c

```
dutil.c: In function ‘NC_mktmp’:
dutil.c:214:9: warning: unused variable ‘tries’ [-Wunused-variable]
  214 |     int tries;
      |         ^~~~~

```

```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I.. -I../include -I../include -I../libnczarr -I/usr/local/hdf5-1.12.1/include -Wall -g -fno-strict-aliasing -MT libnetcdf3_la-v1hpg.lo -MD -MP -MF .deps/libnetcdf3_la-v1hpg.Tpo -c v1hpg.c  -fPIC -DPIC -o .libs/libnetcdf3_la-v1hpg.o
ncio.c:183:1: warning: ‘urlmodetest’ defined but not used [-Wunused-function]
  183 | urlmodetest(const char* path)
      | ^~~~~~~~~~~

```

```
test_pathcvt.c: In function ‘main’:
test_pathcvt.c:133:6: warning: unused variable ‘inputkind’ [-Wunused-variable]
  133 |  int inputkind = NCgetinputpathkind(test->test);
      |      ^~~~~~~~~
At top level:
test_pathcvt.c:192:1: warning: ‘kind2string’ defined but not used [-Wunused-function]
  192 | kind2string(int kind)
      | ^~~~~~~~~~~
```

```
tst_diskless6.c: In function ‘exists’:
tst_diskless6.c:135:9: warning: unused variable ‘stat’ [-Wunused-variable]
  135 |     int stat = 0;
      |         ^~~~
depbase=`echo test_write.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\

```

```
mv -f $depbase.Tpo $depbase.Po
tst_quantize.c: In function ‘main’:
tst_quantize.c:553:22: warning: unused variable ‘dfout’ [-Wunused-variable]
  553 |             union DU dfout;
      |                      ^~~~~

```

--- a/libdispatch/dutil.c
+++ b/libdispatch/dutil.c
@@ -211,10 +211,11 @@ NC_mktmp(const char* base)
     int fd = -1;
     char* tmp = NULL;
     size_t len;
-    int tries;
 #define MAXTRIES 4
 #ifdef HAVE_MKSTEMP
     mode_t mask;
+#else
+    int tries;
 #endif
 
     len = strlen(base)+6+1;

--- a/libsrc/ncio.c
+++ b/libsrc/ncio.c
@@ -49,7 +49,9 @@ extern int ffio_open(const char*,int,off_t,size_t,size_t*,void*,ncio**,void** co
      extern int memio_open(const char*,int,off_t,size_t,size_t*,void*,ncio**,void** const);
 
 /* Forward */
+#ifdef ENABLE_BYTERANGE
 static int urlmodetest(const char* path);
+#endif
 
 int
 ncio_create(const char *path, int ioflags, size_t initialsz,
@@ -179,6 +181,7 @@ NC_HTTP => byterange
 NC_S3SDK => s3
 0 => Not URL
 */
+#ifdef ENABLE_BYTERANGE
 static int
 urlmodetest(const char* path)
 {
@@ -195,3 +198,4 @@ urlmodetest(const char* path)
     ncurifree(uri);
     return kind;
 }
+#endif

--- a/nc_test/tst_diskless6.c
+++ b/nc_test/tst_diskless6.c
@@ -132,7 +132,6 @@ fail(int ret)
 void
 exists(const char* file)
 {
-    int stat = 0;
     FILE* f = NCfopen(file, "r");
     if(f == NULL) fail(NC_EPERM);
     fclose(f);

--- a/nc_test4/tst_quantize.c
+++ b/nc_test4/tst_quantize.c
@@ -550,7 +550,7 @@ main(int argc, char **argv)
             /* union FU fout; */
             union FU xpect[DIM_LEN_5];
             union DU dfin;
-            union DU dfout;
+            /* union DU dfout; */
             union DU double_xpect[DIM_LEN_5];
 	    xpect[0].u = 0x3f8e3000;
 	    xpect[1].u = 0x3f800fff;

--- a/unit_test/test_pathcvt.c
+++ b/unit_test/test_pathcvt.c
@@ -108,7 +108,9 @@ static Test PATHTESTS[] = {
 char* macros[128];
 
 /*Forward */
+#ifdef DEBUG
 static const char* kind2string(int kind);
+#endif
 static char* expand(const char* s);
 static void setmacros(void);
 static void reclaimmacros(void);
@@ -130,7 +132,9 @@ main(int argc, char** argv)
 
     /* Test localkind X path-kind */
     for(test=PATHTESTS;test->test;test++) {
+#ifdef DEBUG
 	int inputkind = NCgetinputpathkind(test->test);
+#endif
         /* Iterate over the test paths */
         for(k=0;k<NKINDS;k++) {
 	    int kind = kinds[k];
@@ -188,6 +192,7 @@ main(int argc, char** argv)
     return (failcount > 0 ? 1 : 0);
 }
 
+#ifdef DEBUG
 static const char*
 kind2string(int kind)
 {
@@ -206,6 +211,7 @@ kind2string(int kind)
     }
     return "Unknown";
 }
+#endif
 
 static char*
 expand(const char* s)

